python/lsst/analysis/ap/zooniverseCutouts.py

```diff
@@ -0,0 +1,371 @@
+                                                          int(source['ccd']),
+                                                          int(source['visit']))
+            image = self.generate_image(science, template, difference, center)
+            with open(self._make_path(source, outputPath), "wb") as outfile:
+                outfile.write(image.getbuffer())
+            count += 1
+        return count
+
+    def generate_image(self, science, template, difference, center):
+        """Get a 3-part cutout image to save to disk, for a single source.
+
+        Parameters
+        ----------
+        science : `lsst.afw.image.ExposureF`
+            Science exposure to include in the cutout.
+        template : `lsst.afw.image.ExposureF`
+            Matched template exposure to include in the cutout.
+        difference : `lsst.afw.image.ExposureF`
+             Matched science minus template exposure to include in the cutout.
+        center : `lsst.geom.SpherePoint`
+            Center of the source to be cut out of each image.
+
+        Returns
+        -------
+        image: `io.BytesIO`
+            The generated image, to be output to a file or displayed on screen.
+        """
+        size = lsst.geom.Extent2I(self.config.size, self.config.size)
+        return self._plot_cutout(science.getCutout(center, size),
+                                 template.getCutout(center, size),
+                                 difference.getCutout(center, size)
+                                 )
+
+    def _plot_cutout(self, science, template, difference):
+        """Plot the cutouts for a source in one image.
+
+        Parameters
+        ----------
+        science : `lsst.afw.image.ExposureF`
+            Cutout Science exposure to include in the image.
+        template : `lsst.afw.image.ExposureF`
+            Cutout template exposure to include in the image.
+        difference : `lsst.afw.image.ExposureF`
+             Cutout science minus template exposure to include in the image.
+
+        Returns
+        -------
+        image: `io.BytesIO`
+            The generated image, to be output to a file via
+            `image.write(filename)` or displayed on screen.
+        """
+        import matplotlib.pyplot as plt
+        from matplotlib import cm
+        import astropy.visualization as aviz
+
+        # TODO: how do we color masked pixels (including edges)?
+
+        def plot_one_image(ax, data, name):
+            """Plot a normalized image on an axis."""
+            if name == 'Difference':
+                norm = aviz.ImageNormalize(data, interval=aviz.ZScaleInterval(),
+                                           stretch=aviz.LinearStretch())
+            else:
+                norm = aviz.ImageNormalize(data, interval=aviz.MinMaxInterval(),
+                                           stretch=aviz.AsinhStretch(a=0.1))
+            ax.imshow(data, cmap=cm.bone, interpolation="none", norm=norm)
```

Comment:
What do you mean "same scale"? The same min,max,stretch? We could, but I don't know whether that is what we actually want or not. Are the pixel values in the coadd and template calibrated to the same flux scale (nanojansky, I'd hope)?

I've asked Eric about followup ticket(s) about tweaking what the images here look like. We'd want to have a collection of a variety of real sources (including a variety of bad ones), since I've found it really hard to get a "good" scaling that covers most cases.

Reply:
Yes, that's what I meant -- same value in source image maps to same RGB color, for all values.

I saw [DM-32014](https://jira.lsstcorp.org/browse/DM-32014), so that just leaves the question of which code is responsible for deciding the `stretch`.

Reply:
I'll leave all of the image formatting questions for that ticket.